go/libraries/doltcore/sqle/dtables/branches_table.go

```diff
@@ -88,23 +106,37 @@ type BranchItr struct {
 }
 
 // NewBranchItr creates a BranchItr from the current environment.
-func NewBranchItr(sqlCtx *sql.Context, ddb *doltdb.DoltDB) (*BranchItr, error) {
-	branches, err := ddb.GetBranches(sqlCtx)
+func NewBranchItr(ctx *sql.Context, ddb *doltdb.DoltDB, all bool) (*BranchItr, error) {
+	var branchRefs []ref.DoltRef
+	var err error
 
-	if err != nil {
-		return nil, err
+	if all {
+		branchRefs, err = ddb.GetRefsOfType(ctx, map[ref.RefType]struct{}{ref.BranchRefType: {}, ref.RemoteRefType: {}})
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		branchRefs, err = ddb.GetBranches(ctx)
+		if err != nil {
+			return nil, err
+		}
 	}
 
-	branchNames := make([]string, len(branches))
-	commits := make([]*doltdb.Commit, len(branches))
-	for i, branch := range branches {
-		commit, err := ddb.ResolveCommitRef(sqlCtx, branch)
+	branchNames := make([]string, len(branchRefs))
+	commits := make([]*doltdb.Commit, len(branchRefs))
+	for i, branch := range branchRefs {
+		commit, err := ddb.ResolveCommitRef(ctx, branch)
 
 		if err != nil {
 			return nil, err
 		}
 
-		branchNames[i] = branch.GetPath()
+		if branch.GetType() == ref.RemoteRefType {
+			branchNames[i] = "  " + "remotes/" + branch.GetPath()
```

Comment:
Why the leading whitespace? Seems like it would just need to be trimmed off for the value to be used.

Reply:
Copypasta error